issuetracker/src/main/java/org/openjdk/skara/issuetracker/jira/JiraProject.java

```diff
@@ -523,10 +506,14 @@ private List<IssueTrackerIssue> queryIssues(String jql) {
 
             if (count < issues.get("total").asInt()) {
                 startAt += issues.get("issues").asArray().size();
-                issues = request.get("search")
-                        .param("jql", jql)
-                        .param("startAt", String.valueOf(startAt))
-                        .execute();
+                req = request.get("search")
+                             .param("jql", jql)
+                             .param("startAt", String.valueOf(startAt));
+                if (limit > 0) {
+                    req = req.param("maxResults", Integer.toString(limit));
```

Comment:
Is the limit supposed to be the total limit or max size per page (would be great with javadoc specifying this btw)? The `maxResults` parameter is results returned per page. I would think a global limit to be more useful.

Reply:
Yeah, it was meant to be a global limit (we only have one caller currently passing a limit other than -1, and that caller passes 1, so it would have worked, but I agree that the current code is wrong). Will update!